flang/lib/Semantics/resolve-directives.cpp

```diff
@@ -1591,6 +1593,20 @@ bool OmpAttributeVisitor::Pre(
   default:
     break;
   }
+  if (standaloneDir.v == llvm::omp::Directive::OMPD_scan) {
+    if ((std::get<parser::OmpClauseList>(x.t).v.size() != 1)) {
+      context_.Say(standaloneDir.source,
+          "Exactly one of `exclusive` or `inclusive` clause is expected"_err_en_US);
+    }
+    if (!parentContext ||
+        (llvm::omp::getDirectiveAssociation(parentContext->directive) !=
```

Comment:
Taskloop has loop association but it's not mentioned in the error message. I think either the message would have to be updated or the check would have to exclude it.

Reply:
I added a directive set. I copied this piece directly from clang. However, this would be a great improvement so I made the change accordingly. Thank you!

Reply:
If clang was checking this incorrectly and it's not a lot of effort to fix, maybe it's worth creating a small PR to address it there too. Or creating an issue for it with a small reproducer if you're not sure how to fix it yourself.

Reply:
Sure, will look into it.